operator/pkg/common/extensions/mapkubeapis.go

```diff
@@ -80,3 +81,19 @@ func (r *mapKubeAPIsExtensionRun) Execute() error {
 
 	return nil
 }
+
+// AddSelectorOptionIfNeeded conditionally adds label selector to opts for reconciler
+func AddSelectorOptionIfNeeded(selector string, opts []pkgReconciler.Option) ([]pkgReconciler.Option, error) {
```

Comment:
This should be in another file, not `mapkubeapis`

Reply:
I will add a separate file for it. Not sure it belongs to any existing one